pandas/core/strings.py

```diff
@@ -2325,12 +2206,133 @@ def cat(self, others=None, sep=None, na_rep=None, join=None):
             res = Series(res, index=data.index, name=self._orig.name)
         return res
 
-    @copy(str_split)
+    _shared_docs['str_split'] = ("""
+    Split strings around given separator/delimiter.
+
+    Splits the string in the Series/Index from the %(side)s,
+    at the specified delimiter string.Equivalent to :meth:`str.%(method)s`.
+
+    Parameters
+    ----------
+    pat : str, optional
+        String or regular expression to split on.
+        If not specified, split on whitespace.
+    n : int, default -1 (all)
+        Limit number of splits in output.
+        ``None``, 0 and -1 will be interpreted as return all splits.
+    expand : bool, default False
+        Expand the splitted strings into separate columns.
+
+        * If ``True``, return DataFrame/MultiIndex expanding dimensionality.
+        * If ``False``, return Series/Index, containing lists of strings.
+
+    Returns
+    -------
+    Series, Index, DataFrame or MultiIndex
+        Type matches caller unless ``expand=True`` (see Notes).
+
+    Notes
+    -----
+    The handling of the `n` keyword depends on the number of found splits:
+
+    - If found splits > `n`,  make first `n` splits only
+    - If found splits <= `n`, make all splits
+    - If for a certain row the number of found splits < `n`,
+      append `None` for padding up to `n` if ``expand=True``
+
+    If using ``expand=True``, Series and Index callers return DataFrame and
+    MultiIndex objects, respectively.
+
+    See Also
+    --------
+    %(also)s
```

Comment:
For the `See Also`, we can leave it like you implemented it. But what we've been doing in other docstrings is to just add here all the methods that use this docstring. They will be self-referencing, besides referencing the similar methods, but that's not a being deal.

So, in this case, instead of `%(also)s` we would have something like:

```
split : Splits string at the first occurrence of delimiter.
rsplit : Splits string at the last occurrence of delimiter.
```

Note that the description should end in a period (you don't have it). Also, I find these descriptions somehow misleading, as it seems that `n` is always `1`.

But also, there are other methods that I think it's worth adding here.At least: `Series.str.join`, `str.split` and `str.rsplit`.